### ISSUE TITLE
Single @php Blade directive does not work correctly when followed by a @endphp tag

### ISSUE THREAD

**Issue by marcbelletre** [laravel/framework#47948]
### Laravel Version

10.17.1

### PHP Version

8.2.7

### Description

The documentation states that we can use a single `@php` directive like this:
```blade
@php($counter = 1)
```

It works fine until there is another block of raw PHP that uses the full syntax after.

```blade
@php($index = 'test')

@php echo $index; @endphp
```

<img width="1157" alt="Capture d’écran 2023-08-03 à 16 04 50" src="https://github.com/laravel/framework/assets/6853334/d4af37d3-03fb-4d1e-9404-915d541e9878">

In the ouput file the PHP end tag is missing:

```php
<?php($index = 'test')

@php echo $index; ?>
<?php /**PATH /Users/marcbelletre/Sites/laravel/resources/views/welcome.blade.php ENDPATH**/ ?
```

I also tried with older versions of Laravel (8 & 9) and got the same issue.

### Steps To Reproduce

1. Create a blade template with a single line `@php` directive and a block of raw PHP using the `@php ... @endphp` syntax
2. Add a route that renders the blade template
3. Open the URL in the browser

**Comment by crynobone** [comment]
For such usage it would be best to just use:

```html
@php
$counter = 1;

echo $counter;
@endphp
```

or even

```php
@php($counter = 1)

{{ $counter }}
```

**Comment by crynobone** [comment]
A similar PR has been rejected before: https://github.com/laravel/framework/pull/45390#issuecomment-1369849576

**Comment by marcbelletre** [comment]
Ok but I think there should be a warning about this in the docs if this is a known issue. I struggled for a while because I couldn't see what was wrong in my code before noticing it was coming from a bug in Laravel itself.

I encountered this with [roots/sage](https://github.com/roots/sage) which is a package that makes a bridge between Wordpress and Laravel.

The default templates are using the following syntax:

```blade
@extends('layouts.app')

@section('content')
  @while(have_posts()) @php(the_post())
    ...
  @endwhile
@endsection
```

Because of this bug it is impossible to use any raw PHP tags inside the loop which is quite a problem. The only workaround is to replace the default `@php(the_post())` call to `@php the_post() @endphp`
